release-tools/changelog.py

```diff
@@ -0,0 +1,206 @@
+    specFile = os.path.join(snapdPackagingDir, "opensuse", "snapd.spec")
+    changesFile = os.path.join(snapdPackagingDir, "opensuse", "snapd.changes")
+    rewriteVersionNumberFile(
+        # meh this is terrible, to keep the right indentation level,
+        # prepend the number of spaces we currently have in the file to
+        # the version number
+        specFile,
+        "Version:",
+        "        " + opts.version,
+    )
+
+    # also add a template changelog to the changes file
+    date = datetime.datetime.now().strftime("%a, %d %b %Y %H:%M:%S %z")
+
+    email = maintainer[1]
+    templ = f"""-------------------------------------------------------------------
+{date} - {email}
+
+- Update to upstream release {opts.version}
+
+"""
+
+    # first read the existing changelog lines
+    with open(changesFile, "r") as fh:
+        current = fh.read()
+    with open(changesFile, "w") as fh:
+        fh.write(templ)
+        fh.write(current)
+
+
+def main(opts):
+    thisScript = os.path.realpath(__file__)
+    snapdRootGitDir = os.path.dirname(os.path.dirname(thisScript))
+    snapdPackagingDir = os.path.join(snapdRootGitDir, "packaging")
+
+    # read all the changelog entries, expected to be formatted by snappy-dch
+    newChangelogEntry = opts.changelog.read()
+
+    # verify that the changelog entry lines are all in the right format
+    for lineNumber, line in enumerate(newChangelogEntry.splitlines(), start=1):
+        # each line should start with either 4 spaces, a - and then another
+        # space, or 6 spaces
+        if not line.startswith("    - ") and not line.startswith("      "):
+            raise RuntimeError(f"unexpected changelog line format in line {lineNumber}")
+        if len(line) >= 72:
+            raise RuntimeError(
+                f"line {lineNumber} too long, should wrap properly to next line"
+            )
+
+    # read the name and email of the person running the script using i.e. dch
+    # conventions
+    maintainer = debian.changelog.get_maintainer()
+
+    # first handle all of the debian packaging files
+    for distro in debianishDistros:
+        debianPackagingChangelog = os.path.join(
+            snapdPackagingDir, distro[0], "changelog"
+        )
+        with open(debianPackagingChangelog) as fh:
+            ch = debian.changelog.Changelog(fh)
+
+        # setup a new block
+        ch.new_block(
+            package="snapd",
+            version=opts.version + distro[2],
+            distributions=distro[1],
+            urgency="medium",
+            author=f"{maintainer[0]} <{maintainer[1]}>",
+            date=debian.changelog.format_date(),
+        )
+
+        # add the new changelog entry with our standard header
+        # the spacing here is manually adjusted, the top of the comment is always
+        # the same
+        templ = "\n  * New upstream release, LP: #1926005\n" + newChangelogEntry
```

Comment:
Maybe we should replace this with "LP: xxxxx" ?

Reply:
what do you mean? Like verbatim make this

```suggestion
        templ = "\n  * New upstream release, LP: xxxx\n" + newChangelogEntry
```

or did you mean to just drop the "#" ?

```suggestion
        templ = "\n  * New upstream release, LP: 1926005\n" + newChangelogEntry
```